compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -922,6 +925,26 @@ toDiagnostic file world range = \case
         , _relatedInformation = Nothing
         }
 
+    fileRangeFromMaybeLocation :: Maybe SS.Location -> (NormalizedFilePath, Range)
+    fileRangeFromMaybeLocation mbLocation =
+        fromMaybe (scenarioFile, scenarioRange) $ do
+            location <- mbLocation
+            lfModule <- LF.lookupLocationModule world location
+            filePath <- LF.moduleSource lfModule
+            Just (toNormalizedFilePath' filePath, rangeFromLocation location)
+
+    rangeFromLocation :: SS.Location -> LSP.Range
```

Comment:
completely orthogonal but we should also be able to use the commit location for errors during scenario interpretation that happen in a `submit`. I’ll open an issue.

Reply:
That's reasonable.

Also, I don't know exactly why, but this isn't working right now. I don't know if the commit location is off, or the source location, or something else.

Reply:
Let me know if you want me to take a closer look or need a rubber duck for debugging

Reply:
It looks like the commit location wasn't being set during a submit. I changed it, and now the locations show up correctly. (I think I still have an off by one error in the column, but that's easy to fix.)

Reply:
uh that’s not good, nice job catching this